src/util/psbt/responsibilities.rs

```diff
@@ -0,0 +1,147 @@
+// Rust Bitcoin Library
+// Written in 2020 by
+//   The Rust Bitcoin developers
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! Implementation of PSBT responsibilities according to BIP-174
+
+use std::marker::PhantomData;
+use std::error::Error;
+use std::fmt::{Formatter, Display};
+use secp256k1::{Message, Signature};
+use util::bip32::{Fingerprint, DerivationPath};
+use super::PartiallySignedTransaction;
+
+/// Trait for a generic PSBT responsibility (decorator+facade)
+pub trait Responsibility where Self: Sized {
+    /// Constructs responsibility decorator/facade from a given PSBT
+    fn from_psbt(psbt: PartiallySignedTransaction) -> Self;
+
+    /// Converts decorator/facade into the raw PSBT object
+    fn into_psbt(self) -> PartiallySignedTransaction;
+
+    /// Returns PSBT representation behind the decorator/facade
+    fn as_psbt(&self) -> &PartiallySignedTransaction;
+}
+
+macro_rules! impl_responsibility {
+    ($ident:ident) => {
+        impl Responsibility for $ident {
+            fn from_psbt(psbt: PartiallySignedTransaction) -> Self {
+                Self { psbt }
+            }
+
+            fn into_psbt(self) -> PartiallySignedTransaction {
+                self.psbt
+            }
+
+            fn as_psbt(&self) -> &PartiallySignedTransaction {
+                &self.psbt
+            }
+        }
+    };
+}
+
+/// PSBT signing verification error
+#[derive(Debug)]
+pub enum VerificationError {
+}
```

Comment:
Are the variants ignored for the purpose of WIP it's meant to be uninhabited?

Reply:
This is just a mockup to agree on the planned API, so no actual data here